modules/stochastic_tools/test/src/functions/nDRosenbrock.C

```diff
@@ -0,0 +1,62 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "nDRosenbrock.h"
+#include "Sampler.h"
+
+registerMooseObject("StochasticToolsTestApp", nDRosenbrock);
+
+InputParameters
+nDRosenbrock::validParams()
+{
+  InputParameters params = GeneralVectorPostprocessor::validParams();
+  params.addRequiredParam<SamplerName>(
+      "sampler", "The Sampler object to use to perform g-function evaluations.");
+  params.addParam<bool>("classify", false, "Flag to turn return binary values.");
+  params.addParam<Real>("limiting_value", 0.0, "True if value exceeds limiting value.");
+  return params;
+}
+
+nDRosenbrock::nDRosenbrock(const InputParameters & parameters)
+  : GeneralVectorPostprocessor(parameters),
+    _sampler(getSampler("sampler")),
+    _classify(getParam<bool>("classify")),
+    _limiting_value(getParam<Real>("limiting_value")),
+    _values(declareVector("g_values"))
+{
```

Comment:
We should add a param error here if `limiting_value` is defined by the user while  `_classify` is `false`.